File: backend/src/main/java/edusecure/edusecure/entity/Grade.java
package edusecure.edusecure.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "grades")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Grade {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false, unique = true)
    private UUID submissionId;

    @Column(name = "grade_value", nullable = false, length = 32)
    private String value;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String feedback;

    @Column(nullable = false)
    private UUID gradedByLecturerId;

    @Column(nullable = false)
    private Instant gradedAt;

    private Instant lastModifiedAt;
}

File: frontend/src/types/grade.ts
export interface GradeResponse {
  id: string
  submissionId: string
  value: string
  feedback: string
  gradedByLecturerId: string
  gradedAt: string
  lastModifiedAt: string | null
}

export interface MyGradeResponse {
  id: string
  submissionId: string
  value: string
  feedback: string
  lastModifiedAt: string | null
}

export interface CreateGradeRequest {
  value: string
  feedback: string
}

export interface UpdateGradeRequest {
  value: string
  feedback: string
}

